packages/server/src/scripts/parse_and_cache_data.js

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env node
+/* eslint-disable */
+
+require("dotenv").config();
+
+const { loadRecordsForUpload } = require('../arpa_reporter/services/records')
+const _ = require("lodash");
+
+const inquirer = require("inquirer");
+
+// This is so named to keep from accidentally typing "knex" out of habit and thus having some
+// queries inadvertently outside the transaction.
+const knexWithoutTransaction = require("../db/connection");
+const { getUpload, cacheRecords } = require("../arpa_reporter/db/uploads");
+
+async function main() {
+    if (!process.env.POSTGRES_URL) {
+        console.error("must specify POSTGRES_URL env variable");
+        return;
+    }
+    console.log(
+        "Parsing and caching upload data with postgres db: ",
+        process.env.POSTGRES_URL
+    );
+
+    const { uploadId } = await inquirer.prompt([
+        {
+            type: "input",
+            name: "uploadId",
+            message: "What is the uuid of the upload you want to parse and cache:",
+        },
+    ])
+
+    console.log("Parsing and caching upload with id:", uploadId);
+
+
+    const upload = await getUpload(uploadId)
+    const records = await loadRecordsForUpload(upload)
+
+    await cacheRecords(upload, records)
+
+    // Now that the data has been written, check that it matches what we parsed from excel
+    const updatedUpload = await getUpload(uploadId)
+
+    if (_.isEqual(records, updatedUpload.parsed_data)) {
+        console.log("Cached data is equivalent to parsed data")
+    } else {
+        diff = compare(records, updatedUpload.parsed_data)
+        console.log("ERROR: Cached and parsed data are not equivalent")
+        console.log(diff)
+    }
```

Comment:
Is this just a spot-check? Do you have a specific reason to believe that the data inserted into the DB might not match what was provided to the insert statement?

Reply:
Just a spot-check, but it has already proved useful: it turns out the json-ification means the timestamps are being treated differently after being read from the cache. The cached timestamps are strings now, where as the excel parser returns them as some other timestamp type

Reply:
Perhaps something like the [cryo library](https://github.com/hunterloftis/cryo) would be useful here, since it has more advanced (de)serialization support for native types but still ultimately serializes to valid JSON?

Reply:
Ooh, this is very interesting and could cause other challenges also, in that it introduces a difference between serialization and deserialization...

I wonder if there's a way for knex and/or pg to set a parser for the data? I see `setTypeParser` on `pg` though this doesn't quite get there / is not this use case per se.

Reply:
I think a quicker solution (and one that matches up with some processing we already have to do in places) is to add a "formatter" in our code that is run on the read-in data and ensures it is in the right format. 

We already have some of these in place to make sure certain fields are strings or numbers, depending on the field (for instance the zipcode looks like a number but has to be treated as a string in case it has a leading zero).

Reply:
neat! I'll explore this parser and make sure it fixes the diffs I'm seeing and then push it to this PR

Reply:
Note that I suspect you will need to use Cryo both to stringify _and_ parse to get the benefit of preserving date objects!